metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -655,6 +676,32 @@ class Compilers(
     }
   }
 
+  private def withPCAndAdjustLsp[T](
+      params: TextDocumentPositionParams,
+      range: LspRange,
+      extractionPos: LspPosition,
+  )(
+      fn: (
+          PresentationCompiler,
+          Position,
+          Position,
+          AdjustLspData,
+      ) => T
+  ): Option[T] = {
+    val path = params.getTextDocument.getUri.toAbsolutePath
+    loadCompiler(path).flatMap { compiler =>
+      val (input, _, adjust) =
+        sourceAdjustments(
+          params,
+          compiler.scalaVersion(),
+        )
+      for {
+        metaRange <- adjust.adjustRange(range).toMeta(input)
+        metaExtractionPos <- adjust.adjustPos(extractionPos).toMeta(input)
+      } yield fn(compiler, metaRange, metaExtractionPos, adjust)
+    }
+  }
+
```

Comment:
You should use`adjustRequest` for input

```suggestion
  private def withPCAndAdjustLsp[T](
      uri: String,
      range: LspRange,
      extractionPos: LspPosition,
  )(
      fn: (
          PresentationCompiler,
          Position,
          Position,
          AdjustLspData,
      ) => T
  ): Option[T] = {
    val path = uri.toAbsolutePath
    loadCompiler(path).flatMap { compiler =>
      val (input, adjustRequest, adjustResponse)=
        sourceAdjustments(
          uri,
          compiler.scalaVersion(),
        )
      for {
        metaRange <- adjustRequest.adjustRange(range).toMeta(input)
        metaExtractionPos <- adjustRequest.adjustPos(extractionPos).toMeta(input)
      } yield fn(compiler, metaRange, metaExtractionPos, adjustResponse)
    }
  }

```